master/internal/api_resourcepool.go

```diff
@@ -203,6 +170,43 @@ func (a *apiServer) ListWorkspacesBoundToRP(
 	}, nil
 }
 
+func (a *apiServer) checkIfPoolIsDefault(poolName string) error {
+	defaultComputePool, err := a.m.rm.GetDefaultComputeResourcePool(a.m.system,
+		sproto.GetDefaultComputeResourcePoolRequest{})
+	if err != nil {
+		return err
+	}
+
+	defaultAuxPool, err := a.m.rm.GetDefaultAuxResourcePool(a.m.system,
+		sproto.GetDefaultAuxResourcePoolRequest{})
+	if err != nil {
+		return err
+	}
+
+	if poolName == defaultComputePool.PoolName ||
+		poolName == defaultAuxPool.PoolName {
+		return errors.Errorf("default resource pool %s cannot be bound to any workspace",
+			poolName)
+	}
+	return nil
+}
+
+func (a *apiServer) canUserModifyWorkspaces(ctx context.Context, ids []int32) error {
+	curUser, _, err := grpcutil.GetUser(ctx)
+	if err != nil {
+		return err
+	}
+
+	if err = workspaceauth.AuthZProvider.Get().CanModifyRPWorkspaceBindings(ctx, *curUser,
+		ids); err != nil {
```

Comment:
lots of indent nits here. i would really just like if we could remove the line length constraint, but that is a battle for another day.